modules/core/03-connection/types/keys.go

```diff
@@ -52,6 +53,9 @@ func IsValidConnectionID(connectionID string) bool {
 
 // ParseConnectionSequence parses the connection sequence from the connection identifier.
 func ParseConnectionSequence(connectionID string) (uint64, error) {
+	if connectionID == exported.LocalhostConnectionID {
```

Comment:
My only hesitation with this fix is that `ParseConnectionSequence("connection-0")` and `ParseConnectionSequence("connection-localhost")` would return the same result

I wonder if we should maintain that calling `ParseConnectionSequence` on the localhost connection is an invalid function call and instead move the if check on the localhost connection id to `IsValidConnectionID` and the genesis.go validation function?